internal/server/server.go

```diff
@@ -101,3 +120,45 @@ func New(port int, cert []byte, key []byte, dbPath string, externalHostname stri
 
 	return s, nil
 }
+
+// reconcileCAStatus checks the status of all CAs in the database and updates their status
+// if necessary.
+func ReconcileCAStatus(dbClient *db.Database, logger *zap.Logger) error {
+	certificateAuthorities, err := dbClient.ListCertificateAuthorities()
+	if err != nil {
+		return fmt.Errorf("failed to list certificate authorities: %w", err)
+	}
+
+	for _, ca := range certificateAuthorities {
+		caDenorm, err := dbClient.GetDenormalizedCertificateAuthority(
+			db.ByCertificateAuthorityDenormalizedID(ca.CertificateAuthorityID),
+		)
+		if err != nil {
+			return fmt.Errorf("failed to get denormalized certificate authority: %w", err)
```

Comment:
In this case, I am not sure we should return the error, but probably only log it and continue. Otherwise, a broken CA that appears early in the list would block the whole process.

Reply:
done!